Sources/XMLCoder/Box/DictionaryBox.swift

```diff
@@ -0,0 +1,89 @@
+//
+//  DictionaryBox.swift
+//  XMLCoderPackageDescription
+//
+//  Created by Vincent Esche on 11/19/18.
+//
+
+import Foundation
+
+// Minimalist implementation of an order-preserving keyed box:
+internal class DictionaryBox {
```

Comment:
same comment on `struct` vs `class` and lack of conformance to Swift standard library protocols as for `ArrayBox`

Reply:
Again, re-reviewing this, I think we could just get away with `extension Dictionary: Box { ... }` if `Box` was a protocol.

Reply:
I chose to keep the API surface area small by only implementing functions that are actually used. There is imho no point in covering all of `Array`/`Dictionary`'s API.